src/parser/deathknight/unholy/modules/features/FesteringStrike.tsx

```diff
@@ -17,49 +18,56 @@ class FesteringStrike extends Analyzer {
     combantants: Combatants,
   };
 
-  constructor(...args) {
-    super(...args);
+  protected enemies!: Enemies;
+  protected combatants!: Combatants;
+
+  constructor(options: Options) {
+    super(options);
 
     this.addEventListener(Events.applydebuffstack.by(SELECTED_PLAYER).spell(SPELLS.FESTERING_WOUND), this.onWoundApply);
     this.addEventListener(Events.removedebuffstack.by(SELECTED_PLAYER).spell(SPELLS.FESTERING_WOUND), this.onWoundRemove);
-    this.addEventListener(Events.removedebuff.by(SELECTED_PLAYER).spell(SPELLS.FESTERING_WOUND), this.onWoundRemove);
+    this.addEventListener(Events.removedebuff.by(SELECTED_PLAYER).spell(SPELLS.FESTERING_WOUND), this.onAllWoundRemove);
     this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.FESTERING_STRIKE), this.onCast);
   }
 
   // used to track how many stacks a target has
-  targets = {};
-
+  targets: number[] = [];
   totalFesteringStrikeCasts = 0;
   festeringStrikeCastsOverThreeStacks = 0;
 
-  onWoundApply(event){
+  onWoundApply(event: ApplyDebuffStackEvent){
     // we only need to look at events applying stacks after the first one, our analysis only cares about times when stacks are >3
     this.targets[event.targetID] = event.stack;
   }
 
-  onWoundRemove(event){
-    this.targets[event.targetID] = event.stack || 0;
+  onWoundRemove(event: RemoveDebuffStackEvent){
+    this.targets[event.targetID] = event.stack;
+  }
+
+  onAllWoundRemove(event: RemoveDebuffEvent){
+    this.targets[event.targetID] = 0;
   }
 
-  onCast(event){
-      this.totalFesteringStrikeCasts += 1;
-      if(this.targets[event.targetID]){
-        const currentTargetWounds = this.targets[event.targetID];
-        if(currentTargetWounds > 3){
-          this.festeringStrikeCastsOverThreeStacks += 1;
-        }
+  onCast(event: CastEvent){
+
+    this.totalFesteringStrikeCasts += 1;
+    if(this.targets[event.targetID!]){
+      const currentTargetWounds = this.targets[event.targetID!];
+      if(currentTargetWounds > 3){
+        this.festeringStrikeCastsOverThreeStacks += 1;
       }
+    }
   }
 
-  suggestions(when) {
+  suggestions(when: When) {
     const percentCastsOverThreeStacks = this.festeringStrikeCastsOverThreeStacks/this.totalFesteringStrikeCasts;
     const strikeEfficiency = 1 - percentCastsOverThreeStacks;
```

Comment:
You calculate strikeEfficiency twice (both statistics and suggestions), you should create a getter for that then to avoid duplicated code.